docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+Gammapy should provide a variety of built-in models, allow to combine existing
+models using limited arithmetics and offer the possibility for users to easily
+implement custom models. A lot of development work has already put into the modeling
+framework of Gammapy. This PIG outlines the missing development work on the model
+framework for Gammapy v1.0.
+
+Proposal
+========
+
+Introduce naming scheme for models
+----------------------------------
+Currently Gammapy implements mainly four categories of models: spectral models,
+morphology or spatial models, combined spectral and spatial models (`SkyModel`)
+and lightcurve or phase models. We propose a uniform naming scheme for these model
+classes, based on the prefixes `Spectral...`, `Spatial...`, `Temporal`:
+
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|   Base class    | Template class                               | IRF folded class     | Model classes               |
++=================+==============================================+======================+=============================+
+|SpectralModel    |SpectralTemplate                              |SpectralIRFModel      | Powerlaw, LogParabola, ...  |
+|(SpectralModel)  |(TableModel)                                  |                      |                             |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|SpatialModel     |SpatialTemplate                               |SpatialIRFModel?      | Shell, SpatialGaussian, ... |
+|(SkySpatialModel)|(SkyDiffuseMap)                               |                      |                             |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|TemporalModel    |PhaseCurveTemplate, LightCurveTemplate        |                      | Sine, ...                   |
+|(None)           |(PhaseCurveTableModel), (LightCurveTableModel)|                      |                             |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|SourceModel      |SourceTemplate                                |SourceIRFModel        |Arithmetic spectral/spatial  |
+|(SkyModel)       |(SkyDiffuseCube)                              |                      |models?                      |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|BackgroundModel  |BackgroundTemplate?                           |BackgroundIRFModel    |Arithmetic background        |
+|(None)           |(None)                                        |      (None)          |models?                      |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+
```

Comment:
I wonder if the scheme here is not a bit too systematic. `SourceIRFModel` and `BackgroundIRFModel` are the same thing in a sense. They contain a `MapGeom` and have a `.npred` method. 

Isn't the relevant point here, the geometry? 

A convolved model is meant to return the `npred` on a given geometry (1D, 2D or 3D). Therefore we could rather have a `MapModel` that can be either an `IRFMapModel` or a `TemplateMapModel`.

Doing so might also alleviate the issue of `SpatialIRFModel` since a `Map` can be 2D or 3D.

Reply:
Just to clarify: `SourceIRFModel` and `BackgroundModel` are not the same thing. Yes, both return the predicted number of counts, but the purpose of the `SourceIRFModel` is to bundle the (differential) model component with it's corresponding IRFs., while the `BackgroundModel` represents an already integrated model, which just introduces additional scaling  parameters for norm and tilt. I agree the `BackgroundModel` could be implemented using a more general `MapModel` object...I'll re-think this part.